### DISCUSSION TITLE
[Suggestions] Game Settings, new option which allow to make a specific automatic save just after launching a scenario

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#7585] [Ideas]
One option I would like to see in fheroes2 is the possibility to make a specific automatic save just after launching a scenario.
"Game: Create automatic save at start of scenario"

**Comment by Branikolog** [comment]
Can you clarify, how should it work?

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> Can you clarify, how should it work?

Once the program has performed all its tasks for the creation of the scenario, random choices for spells in mage guilds, shrines, pyramids, for creatures on the adventure map, for castles and buildings, for artifacts, for resources, etc., etc., and just before the player gets the hand, so also just before the display of the global events messages if any, the program creates an automatic save file (in the background, a bit like an autosave file but just once at the beginning of the scenario) with a specific name, for example "scenario_start_scenario name" to allow to have a precise save of the beginning of the scenario to have the random choices of the program saved which can be useful for replaying exactly the same scenario and also for testing the scenario as needed.

**Comment by Branikolog** [comment]
Humm... I can hardly imagine the conditions player would like to use such save... Probably in a very rare cases.... Should we really make such special options for saves...?

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> Humm... I can hardly imagine the conditions player would like to use such save... Probably in a very rare cases.... Should we really make such special options for saves...?

The first thing I do when I run a scenario is to make a save of the beginning of the scenario in case I want to replay the same scenario with exactly the same playing conditions.
And also in case I would need the save file of the beginning of the scenario to report bugs. :)

Now if the program could do it automatically.
I don't think it takes a lot of time to develop. :)

**Comment by ihhub** [comment]
I am moving this issue to the discussion section. Such feature is not going to be used by the majority of players.
